src/control/cmd/daos/container.go

```diff
@@ -629,8 +638,9 @@ func (cmd *containerStatCmd) Execute(_ []string) error {
 
 func printContainerInfo(out io.Writer, ci *containerInfo, verbose bool) error {
 	epochs := ci.SnapshotEpochs()
+
 	epochStrs := make([]string, *ci.NumSnapshots)
-	for i := uint32(0); i < *ci.NumSnapshots; i++ {
+	for i := uint32(0); ((epochs != nil) && (i < *ci.NumSnapshots)); i++ {
```

Comment:
In hindsight, I think it would be a bit less complicated and error-prone to key this off of the length of the epochs slice (in Go, a nil slice is evaluated as a zero-length slice of whichever type was defined).

e.g.
```go
	epochStrs := make([]string, len(epochs))
	for i, e := range epochs {
		epochStrs[i] = fmt.Sprintf("%d", e)
	}
```

This way, you don't need the extra nil check and it simplifies the loop logic. If epochs is an empty slice, then there's nothing to range over.